utils/cloudfront.go

```diff
@@ -259,6 +260,33 @@ func (d *Distribution) Get(distId string) (*cloudfront.Distribution, error) {
 	return resp.Distribution, nil
 }
 
+func (d *Distribution) SetCertificateAndCname(distId, certId string, domains []string) error {
+	resp, err := d.Service.GetDistributionConfig(&cloudfront.GetDistributionConfigInput{
+		Id: aws.String(distId),
+	})
+	if err != nil {
+		return err
+	}
+
+	aliases := d.getAliases(domains)
+	DistributionConfig, ETag := resp.DistributionConfig, resp.ETag
+	DistributionConfig.Aliases = aliases
+
+	DistributionConfig.ViewerCertificate.Certificate = aws.String(certId)
+	DistributionConfig.ViewerCertificate.IAMCertificateId = aws.String(certId)
+	DistributionConfig.ViewerCertificate.CertificateSource = aws.String("iam")
+	DistributionConfig.ViewerCertificate.SSLSupportMethod = aws.String("sni-only")
+	DistributionConfig.ViewerCertificate.MinimumProtocolVersion = aws.String("TLSv1.2_2018")
```

Comment:
Should this be hardcoded? Is this the same `MinimumProtocolVersion` we have at the moment?

Reply:
It does match what we set in https://github.com/alphagov/paas-cdn-broker/pull/12/files#diff-9ce0b3b0fd6c33664bdb1b8f843fa74eR276. I'll see about having that in one place.

Reply:
Ah, this function entirely replaces the `SetCertificate` where the other occurence is. Will remove `SetCertificate`.